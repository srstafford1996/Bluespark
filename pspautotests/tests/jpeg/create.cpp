#include <common.h>
#include <pspjpeg.h>
#include <psputility.h>

extern "C" int main(int argc, char *argv[]) {
	sceUtilityLoadModule(PSP_MODULE_AV_AVCODEC);

	checkpointNext("Basic:");
	checkpoint("  Before init: %08x", sceJpegCreateMJpeg(480, 272));
	checkpoint("  Init: %08x", sceJpegInitMJpeg());
	checkpoint("  After init: %08x", sceJpegCreateMJpeg(480, 272));
	checkpoint("  Twice: %08x", sceJpegCreateMJpeg(480, 272));
	checkpoint("  Delete: %08x", sceJpegDeleteMJpeg());
	checkpoint("  After delete: %08x", sceJpegCreateMJpeg(480, 272));
	sceJpegDeleteMJpeg();

	checkpointNext("Widths:");
	static const int sizes[] = { 0x80000000, 0xFFFF0000, 0xFFFF8000, -1, 0, 1, 2, 3, 4, 5, 480, 512, 1023, 1024, 1025, 2048, 0x8000, 0x7FFFFFFF};
	for (size_t i = 0; i < ARRAY_SIZE(sizes); ++i) {
		char temp[256];
		checkpoint("  Width %d: %08x", sizes[i], sceJpegCreateMJpeg(sizes[i], 272));
		sceJpegDeleteMJpeg();
	}

	checkpointNext("Heights:");
	for (size_t i = 0; i < ARRAY_SIZE(sizes); ++i) {
		char temp[256];
		checkpoint("  Height %d: %08x", sizes[i], sceJpegCreateMJpeg(480, sizes[i]));
		sceJpegDeleteMJpeg();
	}

	checkpointNext("Total:");
	checkpoint("  Size 1024x1024: %08x", sceJpegCreateMJpeg(1024, 1024));
	sceJpegDeleteMJpeg();

	sceUtilityUnloadModule(PSP_MODULE_AV_AVCODEC);
	return 0;
}