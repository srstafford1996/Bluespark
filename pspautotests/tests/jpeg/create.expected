[x] Basic:
[x]   Before init: 80650039
[r]   Init: 00000000
[x]   After init: 00000000
[x]   Twice: 80650039
[x]   Delete: 00000000
[x]   After delete: 00000000

[x] Widths:
[x]   Width -2147483648: 00000000
[x]   Width -65536: 00000000
[x]   Width -32768: 00000000
[x]   Width -1: 00000000
[x]   Width 0: 00000000
[x]   Width 1: 00000000
[x]   Width 2: 00000000
[x]   Width 3: 00000000
[x]   Width 4: 00000000
[x]   Width 5: 00000000
[x]   Width 480: 00000000
[x]   Width 512: 00000000
[x]   Width 1023: 00000000
[x]   Width 1024: 00000000
[x]   Width 1025: 80650020
[x]   Width 2048: 80650020
[x]   Width 32768: 80650020
[x]   Width 2147483647: 80650020

[x] Heights:
[x]   Height -2147483648: 00000000
[x]   Height -65536: 00000000
[x]   Height -32768: 00000000
[x]   Height -1: 00000000
[x]   Height 0: 00000000
[x]   Height 1: 00000000
[x]   Height 2: 00000000
[x]   Height 3: 00000000
[x]   Height 4: 00000000
[x]   Height 5: 00000000
[x]   Height 480: 00000000
[x]   Height 512: 00000000
[x]   Height 1023: 00000000
[x]   Height 1024: 00000000
[x]   Height 1025: 00000000
[x]   Height 2048: 00000000
[x]   Height 32768: 00000000
[x]   Height 2147483647: 00000000

[x] Total:
[x]   Size 1024x1024: 00000000
